tests/metrics/annotation_util.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"sync"
+)
+
+//Client request grafana API on a set of resource paths.
+type client struct {
+	// base is the root URL for all invocations of the client
+	baseUrl url.URL
+	client  *http.Client
+}
+
+//Annotation is a specification of the desired behavior of adding annotation
+type Annotation struct {
+	DashboardId         int      `json: "dashboardId"`
+	PanelId             int      `json: "panelId"`
```

Comment:
`DashboardId` and `PanelId` are optional, so we should use `*int` here and add inform serializer to omit these field is empty (nil), the notation is `json:"dashboardId,omitempty"`

Reply:
So do `tags`.

Reply:
dashboardId和pannelId明显是不能忽略的啊，要不然知道加哪张图上
tags accept

Reply:
The scenario is that users may add the global annotation (without `DashboardId` and `PanelId`) to grafana and query it in multiple dashboards. To me it is the common case because when I am coding I cannot ensure the grafana dashboard id and panel id is consistent in different clusters and different e2e tests or stability tests.

Reply:
can we post an annotation without  dashboardID and panelID?

Reply:
Yes, https://grafana.com/docs/http_api/annotations/#create-annotation

Reply:
We can leave them optional for a wider range of uses. @qiffang

Reply:
From the original requirement,  we want to add annotation in a graph(panel). 
Of course, we can support it.
Accept